src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs

```diff
@@ -398,12 +401,35 @@ public override void ExecuteCmdlet()
                    }
 
                    HandleActions();
-                   var result = (PSAzureProfile) (task.ConfigureAwait(false).GetAwaiter().GetResult());
-                   WriteObject(result);
+
+                   try
+                   {
+                       var result = (PSAzureProfile)(task.ConfigureAwait(false).GetAwaiter().GetResult());
+                       WriteObject(result);
+                   }
+                   catch (AuthenticationFailedException ex)
+                   {
+                       if(IsUableToOpenWebPageError(ex))
+                       {
+                           WriteWarning(Resources.InteractiveAuthNotSupported);
+                           WriteDebug(ex.ToString());
+                       }
+                       else
+                       {
+                           WriteWarning(Resources.SuggestToUseDeviceCodeAuth);
+                           throw;
+                       }
+                   }
                });
             }
         }
 
+        private bool IsUableToOpenWebPageError(AuthenticationFailedException exception)
+        {
+            return exception.InnerException is MsalClientException && ((MsalClientException)exception.InnerException)?.ErrorCode == MsalError.LinuxXdgOpen
+                            || (exception.Message?.ToLower()?.Contains("uable to open a web page") ?? false);
```

Comment:
A typo. Original message: "Unable to open a web page using xdg-open."

```suggestion
                            || (exception.Message?.ToLower()?.Contains("unable to open a web page") ?? false);
```

Reply:
Good catch, I just noticed as well, thx.